refactor: fix mypy errors for checkers/a*

Fix `mypy` errors for the following:

```
cve_bin_tool/checkers/atftp.py:15: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/atftp.py:16: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/asterisk.py:18: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/asterisk.py:19: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/assimp.py:15: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/assimp.py:16: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/asn1c.py:15: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/apcupsd.py:16: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/apcupsd.py:17: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/acpid.py:17: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/acpid.py:18: error: Need type annotation for "FILENAME_PATTERNS" (hint: "FILENAME_PATTERNS: List[<type>] = ...")
cve_bin_tool/checkers/accountsservice.py:18: error: Need type annotation for "CONTAINS_PATTERNS" (hint: "CONTAINS_PATTERNS: List[<type>] = ...")

```

Helpful resources:

- [Python typing documentation](https://docs.python.org/3/library/typing.html)
- [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)
- If you can't find the answer in the docs, googling the error messages above will often help!
- [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md)
- If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)

Tiny tips for new contributors:

- cve-bin-tool uses https://www.conventionalcommits.org/ style for commit messages, and we have a test that checks the title of your PR. A good potential title for this one is in the title of this issue.
- You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR comments where ISSUENUMBER is the actual number of the issue.

--- a/cve_bin_tool/checkers/accountsservice.py
+++ b/cve_bin_tool/checkers/accountsservice.py
@@ -11,11 +11,13 @@ https://www.cvedetails.com/product/48266/Freedesktop-Accountsservice.html?vendor
 https://www.cvedetails.com/product/63450/Accountsservice-Project-Accountsservice.html?vendor_id=21123
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class AccountsserviceChecker(Checker):
-    CONTAINS_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
     FILENAME_PATTERNS = [r"accountsservice"]
     VERSION_PATTERNS = [r"accounts-daemon ([0-9]+\.[0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [

--- a/cve_bin_tool/checkers/acpid.py
+++ b/cve_bin_tool/checkers/acpid.py
@@ -10,12 +10,14 @@ https://www.cvedetails.com/product/21292/Tedfelix-Acpid.html?vendor_id=11569
 https://www.cvedetails.com/product/23004/Tedfelix-Acpid2.html?vendor_id=11569
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class AcpidChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = ["acpid\\-([0-9]+\\.[0-9]+\\.[0-9]+)"]
     VENDOR_PRODUCT = [
         ("tedfelix", "acpid"),

--- a/cve_bin_tool/checkers/apcupsd.py
+++ b/cve_bin_tool/checkers/apcupsd.py
@@ -9,11 +9,13 @@ https://www.cvedetails.com/product/1160/APC-Apcupsd.html?vendor_id=625
 https://www.cvedetails.com/product/55361/Apcupsd-Apcupsd.html?vendor_id=16632
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class ApcupsdChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = ["apcupsd ([0-9]+\\.[0-9]+\\.[0-9]+)"]
     VENDOR_PRODUCT = [("apc", "apcupsd"), ("apcupsd", "apcupsd")]

--- a/cve_bin_tool/checkers/asn1c.py
+++ b/cve_bin_tool/checkers/asn1c.py
@@ -8,11 +8,13 @@ CVE checker for asn1c
 https://www.cvedetails.com/product/39543/Asn1c-Project-Asn1c.html?vendor_id=16848
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class Asn1CChecker(Checker):
-    CONTAINS_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
     FILENAME_PATTERNS = ["asn1c"]
     VERSION_PATTERNS = [r"asn1c-([0-9]+\.[0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [("asn1c_project", "asn1c")]

--- a/cve_bin_tool/checkers/assimp.py
+++ b/cve_bin_tool/checkers/assimp.py
@@ -8,11 +8,13 @@ CVE checker for assimp
 https://www.cvedetails.com/product/107135/Assimp-Assimp.html?vendor_id=26183
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class AssimpChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"assimp-([0-9]+\.[0-9]+\.[0-9]+)"]
     VENDOR_PRODUCT = [("assimp", "assimp")]

--- a/cve_bin_tool/checkers/asterisk.py
+++ b/cve_bin_tool/checkers/asterisk.py
@@ -11,12 +11,14 @@ https://www.cvedetails.com/product/113194/Sangoma-Asterisk.html?vendor_id=9238
 https://www.cvedetails.com/product/12770/Asterisk-Open-Source.html?vendor_id=6284
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class AsteriskChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [
         r"asterisk-([0-9]+\.[0-9]+\.[0-9]+)",
         r"ast_uuid_init\r?\n([0-9]+\.[0-9]+\.[0-9]+)",

--- a/cve_bin_tool/checkers/atftp.py
+++ b/cve_bin_tool/checkers/atftp.py
@@ -8,11 +8,13 @@ CVE checker for atftp
 https://www.cvedetails.com/product/53662/Atftp-Project-Atftp.html?vendor_id=19769
 
 """
+from typing import List
+
 from cve_bin_tool.checkers import Checker
 
 
 class AtftpChecker(Checker):
-    CONTAINS_PATTERNS = []
-    FILENAME_PATTERNS = []
+    CONTAINS_PATTERNS: List[str] = []
+    FILENAME_PATTERNS: List[str] = []
     VERSION_PATTERNS = [r"([0-9]+\.[0-9]+\.[0-9]+)\r?\natftp"]
     VENDOR_PRODUCT = [("atftp_project", "atftp")]

